app/models/concerns/tagable.rb

```diff
@@ -1,8 +1,15 @@
 module Tagable
   def tag(name_or_id)
-    Tagging.create(tag_id:         Tag.find!(name_or_id)[:id],
-                   tagable_class:  self.class.name,
-                   tagable_id:     self.id)
+    unless has_tag?(name_or_id)
```

Comment:
while there's nothing really wrong with using this has_tag? helper, the 'rails way' of writing this would be:
``` Tagging.find_or_create_by() ```

see: http://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-find_or_create_by